godel_surface_detection/src/detection/surface_detection.cpp

```diff
@@ -297,12 +299,20 @@ namespace godel_surface_detection
       pcl::copyPointCloud(*full_cloud_ptr_, cloud);
     }
 
-
     void SurfaceDetection::get_full_cloud(sensor_msgs::PointCloud2 cloud_msg)
     {
       pcl::toROSMsg(*full_cloud_ptr_, cloud_msg);
     }
 
+    void SurfaceDetection::get_process_cloud(CloudRGB& cloud)
+    {
+      pcl::copyPointCloud(*process_cloud_ptr, cloud);
+    }
+
+    void SurfaceDetection::get_process_cloud(sensor_msgs::PointCloud2 cloud_msg)
+    {
+      pcl::toROSMsg(*process_cloud_ptr, cloud_msg);
```

Comment:
You are copying the point cloud type by value into this function and then copying the contents of `process_cloud_ptr` into _that_ copy. I think you need to pass `cloud_msg` by reference to effectively "return".